internal/shard/shard_thread.go

```diff
@@ -75,9 +77,13 @@ func (shard *ShardThread) runCronTasks() {
 	shard.lastCronExecTime = utils.GetCurrentTime()
 }
 
-func (shard *ShardThread) registerWorker(workerID string, workerChan chan *ops.StoreResponse) {
+func (shard *ShardThread) registerWorker(workerID string, processingChan ...chan *ops.StoreResponse) {
 	shard.workerMutex.Lock()
-	shard.workerMap[workerID] = workerChan
+	if len(processingChan) > 1 {
+		shard.workerMap[workerID] = processingChan[0]        // response channel on first index
+		shard.preProcessingMap[workerID] = processingChan[1] // preprocessing channel on second index
```

Comment:
Do you think it will be clearer if instead of using the spread operator throughout the function call stack we passed a struct with named fields for each of these channels? It would make the code a lot clearer.

Reply:
Made required changes